src/main/kotlin/ecommerce/ProductController.kt

```diff
@@ -0,0 +1,68 @@
+package ecommerce
+
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Controller
+import org.springframework.ui.Model
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import java.net.URI
+
+@Controller
+class ProductController(private val productRepository: ProductRepository) {
+    @PostMapping("/api/products")
+    fun createProduct(
+        @RequestBody product: Product,
+    ): ResponseEntity<Void> {
+        val id = productRepository.create(product)
+        return ResponseEntity.created(URI.create("/api/products/$id")).build()
+    }
+
+    @GetMapping("/api/products")
+    fun readProducts(): ResponseEntity<List<Product>> {
+        val products = productRepository.getAll()
+        return ResponseEntity.ok(products)
+    }
+
+    @PutMapping("/api/products/{id}")
+    fun updateProduct(
+        @RequestBody newProduct: Product,
+        @PathVariable id: Long,
+    ): ResponseEntity<Void> {
+        val updated = productRepository.update(id, newProduct)
+        if (!updated) {
+            throw RuntimeException("Product not found")
+        }
+        return ResponseEntity.ok().build()
+    }
+
+    @DeleteMapping("/api/products/{id}")
+    fun deleteProduct(
+        @PathVariable id: Long,
+    ): ResponseEntity<Void> {
+        val deleted = productRepository.delete(id)
+        if (!deleted) {
+            throw RuntimeException()
+        }
+        return ResponseEntity.noContent().build()
```

Comment:
I think delete checking is not presentation(controller) layer's responsibility. 🤔 
how do u think? what is presentation layer's role?